### DISCUSSION TITLE
Google Auth x Adonis

### DISCUSSION THREAD

**Discussion by Mzzay** [adonisjs/core#1302] [Help]
Hi, I connect users with google authentification in my react-native app. But I have no idea that what store in my db to check : 
1. if it's a new user or not
2. if the user is connected or not

My small ideas : 
1. Store the user's email and check if is existing or not
2. When user login send his data to Adonis and adonis send a jwt token. 

But for the idea 2, how to generate a jwt token, because I have no "password" ? 

( My app is on react native ) 

Thanks.

**Comment by thetutlage** [comment]
Which version of AdonisJS?

**Reply by Mzzay** [reply]
4.1

**Comment by iammrsea** [comment]
Personally, I would go with the second idea. For the password, you can use each user's google id as their password.

**Reply by Mzzay** [reply]
User's google id are private ? Or just accessible if user is logged ?

**Reply by iammrsea** [reply]
I think you can only access it with a valid id_token

**Reply by Mzzay** [reply]
Currently I have it when the user logs into my application

**Comment by Mzzay** [comment]
ok so, I create an user with his email as "email" and id as "password". Then I send to him his jwt token. But in my case, what is the token table for ?

**Reply by iammrsea** [reply]
I guess the token table is to manage (that is, keep track of) token expiration

**Reply by Mzzay** [reply]
yes but when need I to defined it as revoked ?

**Comment by rodrigogaraujo** [comment]
Any ideas for this?
I need to be able to login with google + jwt token.
